Add documentation for stack commands

vdemeester opened the pull request:
We already document _experimental_ commands like `plugin` subcommands, adding some documentation for `stack` subcommands and for the `deploy` command.

Also removes the `-f` flags of bundle to follow the single-letter flags evaluation.
- [ ] Add more example on `tasks` subcommand (like what we do on `service tasks`).
- [ ] What to do about the short flag for `--bundle`.

/cc @thaJeztah @dnephin @aanand @tiborvass @icecrime 

Signed-off-by: Vincent Demeester vincent@sbr.pm

thaJeztah commented on api/client/stack/opts.go:
```diff
@@ -12,9 +12,9 @@ import (
+	flags.StringVar(
 		opt,
-		"bundle", "f", "",
+		"bundle", "",
```
Yeah, still wondering if we should change this to `--file`

@dnephin @aanand wdyt?

dnephin replied:
I'm fine with either

thaJeztah replied:
It feels more consistent with `docker build --file`, and `docker-compose --file`

thaJeztah replied:
But better do that in a separate PR 👍

thaJeztah commented on docs/reference/commandline/deploy.md:
````diff
@@ -0,0 +1,57 @@
+Creating service vossibility-stack_vossibility-collector
+```
+
+We can verify that services were correctly created:
````
s/We can/You can/

Also misses a "the";

```Markdown
You can verify that the services were correctly created:
```

thaJeztah commented on docs/reference/commandline/stack_deploy.md:
````diff
@@ -0,0 +1,59 @@
+Creating service vossibility-stack_vossibility-collector
+```
+
+We can verify that services were correctly created:
````
Same here :)

thaJeztah commented:
some nits/questions, but otherwise LGTM

vdemeester committed:
Add documentation for stack commands

Also removes the `-f` flags of bundle to follow the single-letter flags
evaluation.

Signed-off-by: Vincent Demeester <vincent@sbr.pm>

icecrime commented:
LGTM

cpuguy83 commented:
LGTM
